dags/deletes.py

```
@@ -536,6 +544,54 @@ def load_and_verify_adhoc_event_deletes_dictionary(
     return dictionary
 
 
+@dagster.op
+def delete_custom_events(
+    context: dagster.OpExecutionContext,
+    cluster: dagster.ResourceParam[ClickhouseCluster],
+    load_pending_deletions: PendingDeletesTable,
+) -> PendingDeletesTable:
+    """Delete events for custom deletions using their predicates."""
+
+    # Get custom deletions from the pending deletions table - ensure team_id is always present
+    custom_deletions_query = f"""
+        SELECT team_id, key
+        FROM {load_pending_deletions.qualified_name}
+        WHERE deletion_type = {DeletionType.Custom}
+        AND team_id IS NOT NULL
+    """
+
+    custom_deletions = cluster.any_host_by_role(
+        lambda client: client.execute(custom_deletions_query), NodeRole.DATA
+    ).result()
+
+    if not custom_deletions:
+        context.add_output_metadata(
+            {"custom_events_deleted": dagster.MetadataValue.int(0), "message": "No custom deletions found"}
+        )
+        return load_pending_deletions
+
+    context.add_output_metadata({"custom_deletions_count": dagster.MetadataValue.int(len(custom_deletions))})
+
+    # Process each custom deletion with its predicate - team_id is guaranteed to be present
+    for team_id, predicate in custom_deletions:
+        if not team_id or not predicate:
+            context.log.warning(
+                f"Skipping custom deletion with missing team_id or predicate: team_id={team_id}, predicate={predicate}"
+            )
+            continue
```

If this branch hits it seems like a serious bug and something that should probably throw an exception for somebody to investigate instead of just dropping the request

I think these would also get marked as completed but I haven't really dug in to check thoroughly